### DISCUSSION TITLE
关于JuiceFS传染性的问题

### DISCUSSION THREAD

**Discussion by Quan-HS** [juicedata/juicefs#3300] [Q&A] [answered]
我们一套AI代码在同时适配S3、GCS、Azure、OSS时，涉及到pandas、tf等改造，目前寻求JuiceFS来提供屏蔽底层对象存储适配的功能，但是JuiceFS有自己的文件系统，那么大量对象存储的数据需要迁移，并且要求数仓跟着改，涉及的范围非常大（数据量大、数仓任务多），请问JuiceFS在不能大规模铺开的情况下如何能平滑的接入？

**Comment by xiaogaozi** [comment, marked as answer]
如果你们还处在评估 JuiceFS 的阶段，建议按业务分批迁移数据。对于历史数据，可以先拷贝一部分到 JuiceFS 中；对于数仓产生的新数据，可以先让数仓任务双写一段时间（即同时写到对象存储和 JuiceFS）。等评估完成以后再让数仓只写到 JuiceFS 中，并删除对象存储上的历史数据。

**Reply by Quan-HS** [reply]
感谢
